model/cluster_discovery.go

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package model
+
+import (
+	"encoding/json"
+	"io"
+	"os"
+)
+
+const (
+	CDS_OFFLINE_AFTER_MILLIS = 1000 * 60 * 30 // 30 minutes
+	CDS_TYPE_APP             = "mattermost_app"
+)
+
+type ClusterDiscovery struct {
+	Id          string `json:"id"`
+	Type        string `json:"type"`
+	ClusterName string `json:"cluster_name"`
+	Hostname    string `json:"hostname"`
+	GossipPort  int32  `json:"gossip_port"`
```

Comment:
This is so different servers can have different ports if needed.  Although we don't use this for Kubes, it was one of the complaints of the old system.